File: hs/5.hs
-- 232792560
nn = 20

tryReduce ds n d = if m == 0 && isDivisible then tryReduce ds q d else n
    where (q,m) = n `divMod` d
          isDivisible = all (\x -> q `mod` x == 0) ds

smallestMult n = foldl (tryReduce ds) p ds
    where p = product ds
          ds = [2..n]

main = putStrLn $ show $ smallestMult nn

File: py/5.py
# 232792560
import euler
D = [n for n in xrange(2,21)]
n = euler.product(D)
m = 0
for d in D:
    nx = n / d
    while all([nx % dx == 0 for dx in D]):
        n = nx
        nx = n / d
        m = n
print m
